src/Raven.Server/Documents/ETL/Providers/Queue/Kafka/KafkaEtl.cs

```diff
@@ -116,13 +123,33 @@ void ReportHandler(DeliveryReport<string, byte[]> report)
                 }
             }
         }
+        if (itemsPerTopic.Count == 0)
+            return count;
+        QueueWithItems<KafkaItem> currentTopic = itemsPerTopic[0];
+        
+        
+        try
+        {
+            _producer.BeginTransaction();
+        }
+        catch (KafkaException e)
+        {
+            var errorMessageBuilder = new StringBuilder($"ETL process: {Name}. Begin transaction failed.");
+            if (TryToGetAclErrorsDetails(currentTopic.Name, out var detailsMessage))
```

Comment:
At this stage you're not working with topics yet. Why do we pass `currentTopic.Name`?

Reply:
Good point. Also, we don't work with topics on the `commit transaction` level already.

Reply:
https://github.com/ravendb/ravendb/pull/17577/commits/ba1b9f05728c8fa1942c5e694dbe6d914ba3f5c2